src/client.rs

```diff
@@ -933,6 +934,19 @@ impl Spotify {
         self.convert_result(&result)
     }
 
+    /// Get a list of the songs saved in the current Spotify user's "Your Music"
+    /// library.
+    ///
+    /// [Reference](https://developer.spotify.com/web-api/get-users-saved-tracks/)
+    pub fn current_user_saved_tracks_stream(
```

Comment:
After this PR merged, what's the next step? Planing to add a new method with `_stream` suffix for each endpoint except `current_user_saved_tracks`?

Reply:
That's what I was explaining and questioning in this comment - https://github.com/ramsayleung/rspotify/pull/166#issuecomment-793698024

Seeing as there isn't anyone else giving their opinion I would just come to an agreement between you and I and just use that.

Reply:
I prefer `_stream` suffix to `_auto` suffix, since the latter one is not that self-explanatory as `_stream` suffix, and `stream` is defined as a sequence of elements supporting sequential and parallel aggregate operations in a we-all-know way, such as Java's `Stream` and Rust's Future `stream`.

And let me know what you think now :-)

Reply:
Yes but the thing is that `_stream` might be confusing, as it returns an `Iterator` when sync mode is enabled, and it returns a `Stream` when async mode is enabled. `_auto` works for both, and so does `Paginator`, IMO. Perhaps there's a better alternative than `_auto`, but my concern was that we were mixing up the async implementation with the generalized one.

Reply:
Yes, I get you point! It makes sense. Perhaps there's indeed a better choice than `_auto`. But in this moment, no clue comes to my mind

Reply:
By the way, I am thinking about is it possible to generate the new method with `_stream` or `_auto` suffix by macro, since the code pattern is similar.

Reply:
> By the way, I am thinking about is it possible to generate the new method with _stream or _auto suffix by macro, since the code pattern is similar.

This is non-trivial to implement because of generic types in functions. A macro that takes these into account is quite complicated and not worth the effort in this case, unfortunately.

Reply:
There is another suffix name came to my mind, `_range` suffix, inspiration from C++ [Range library](https://en.cppreference.com/w/cpp/ranges)  :)

Reply:
Here's another idea I added to my comment, which I modified: what if the endpoints are automatically paginated by default, and you can use the `_manual`-suffixed endpoints in case you want more control over them? It's a saner default, as most times you just want an iterator, and we solve the naming issue pretty easily :)

Reply:
Yes, it makes much sense. We could make the endpoints paginated by default, since the developers want iterator at most times as you point out. My concern is that if we change all manual endpoints to be paginated and then add a `_manual` suffix for the original one, then we will change the function signature of all endpoints, it does break the compatibility.